src/main.ts

```diff
@@ -148,212 +98,108 @@ export class Content {
-    repo: repo.name
-  })
-  let parentinfo = ''
-  if (repoinfo.parent?.full_name) {
-    parentinfo = repoinfo.parent.full_name
-  }
-
-  // search for action.yml file in the root of the repo
-  try {
-    const { data: yml } = await client.rest.repos.getContent({
-      owner: repo.owner,
-      repo: repo.name,
-      path: 'action.yml'
-    })
-
-    // todo: warning: duplicated code here
-    if ('name' in yml && 'download_url' in yml) {
-      result.name = yml.name
-      result.owner = repo.owner
-      result.repo = repo.name
-      result.forkedfrom = parentinfo
+): Promise<Content[]> {
+  const actions: Content[] = []
 
-      if (yml.download_url !== null) {
-        result.downloadUrl = yml.download_url
-      }
-    }
-  } catch (error) {
-    core.debug(`No action.yml file found in repository: ${repo.name}`)
+  
+  var searchQuery = '+filename:action+language:YAML';
+  if (username) {
+    core.info(`Search for action files of the user [ ${username} ]`)
+    searchQuery = searchQuery.concat('+user:', username)
   }
 
-  if (result.name === '') {
-    try {
-      // search for the action.yaml, that is also allowed
-      const { data: yaml } = await client.rest.repos.getContent({
-        owner: repo.owner,
-        repo: repo.name,
-        path: 'action.yaml'
-      })
-
-      if ('name' in yaml && 'download_url' in yaml) {
-        result.name = yaml.name
-        result.owner = repo.owner
-        result.repo = repo.name
-        result.forkedfrom = parentinfo
-        if (yaml.download_url !== null) {
-          result.downloadUrl = yaml.download_url
-        }
-      }
-    } catch (error) {
-      core.debug(`No action.yaml file found in repository: ${repo.name}`)
-    }
+  if (organization !== '') {
+    core.info(`Search for action files under the organization [ ${organization} ]`)
+    searchQuery = searchQuery.concat('+org:', organization)
   }
 
-  // todo: ratelimiting can be enabled on GHES as well, but is off by default
-  // we can probably load it from an api call and see if it is enabled, or try .. catch 
-  if (!isEnterpriseServer) {
-    // search API has a strict rate limit, prevent errors
-    var ratelimit = await client.rest.rateLimit.get()
-    if (ratelimit.data.resources.search.remaining <= 2) {
-      // show the reset time
-      var resetTime = new Date(ratelimit.data.resources.search.reset * 1000)
-      core.debug(`Search API reset time: ${resetTime}`)
-      // wait until the reset time
-      var waitTime = resetTime.getTime() - new Date().getTime()
-      if (waitTime < 0) {
-        // if the reset time is in the past, wait 2,5 seconds for good measure (Search API rate limit is 30 requests per minute)
-        waitTime = 2500
-      } else {
-        // back off a bit more to be more certain
-        waitTime = waitTime + 1000
-      }
-      core.info(`Waiting ${waitTime/1000} seconds to prevent the search API rate limit`)
-      await new Promise(r => setTimeout(r, waitTime));
-    }
-  }
```

Comment:
I think this needs to stay, to prevent rate limit issues

Reply:
Added the snippet back.